lms/views/api/canvas/files.py

```diff
@@ -30,12 +30,19 @@ def via_url(self):
         :raise lms.services.CanvasAPIError: if the Canvas API request fails.
             This exception is caught and handled by an exception view.
         """
+        application_instance = self.request.find_service(
+            name="application_instance"
+        ).get()
+
+        module_item_configuration = self.request.find_service(name="assignment").get(
+            application_instance.tool_consumer_instance_guid,
+            self.request.matchdict["resource_link_id"],
+        )
+
         public_url = self.canvas.public_url_for_file(
-            file_id=self.request.matchdict["file_id"],
-            course_id=self.request.matchdict["course_id"],
-            # Teachers can have broad permissions and see files that aren't in
-            # the course. So do this slower check (extra API call) to warn the
-            # teacher that their students might not be able to see the file.
+            module_item_configuration,
+            self.request.matchdict["file_id"],
+            self.request.matchdict["course_id"],
```

Comment:
I think kwargs are easier to keep straight once we start to get lots of arguments (or even not) and form some nice natural commentary / documentation. They also facilitate refactoring.

Reply:
In this case the variable name (`module_item_configuration`) and dict keys (`"file_id"` and `"course_id"`) already make it obvious what each argument is. If kwargs were used they'd just all be repeated:

```python
public_url = self.canvas.public_url_for_file(
    module_item_configuration=module_item_configuration,
    file_id=self.request.matchdict["file_id"],
    course_id=self.request.matchdict["course_id"],
    ...
```